apps/shared/ResourceController.qml

```diff
@@ -95,19 +99,20 @@ Item {
                     _lastStateOwner = data.owner
                 }
                 calculateStatus()
+            } else if (message === "network-enable") {
```

Comment:
if this message is asynchronous, is it worth adding a check for `&& !connectionHelper.online` ?  or maybe attemptToConnectNetwork() already has such a check, internally?  From a quick check, it looks like it does have one for `m_netman->defaultRoute()->state() == "online"` but still does some signal emission etc stuff... I guess it's fine as-is but thought I'd raise it for discussion.

Reply:
Thanks, it's a good point. As you mention, `ConnectionHelper` does do its own check with the fallback being that the connection dialog doesn't open (which is what we want here).

However, correct behaviour will rely on the two checks being equivalent. From digging through the `ConnectionHelper` code it's not 100% clear to me that `NetworkManager->defaultRoute()->state()` will always give exactly the same result as the `NetworkManager::state` property, but I think they must be pretty close, given that updating `NetworkManager::state` will synchronously cause the default route to be updated.

My preference would be to avoid complicating the code here and leave the check to `ConnectionHelper`. But I agree there's a risk, so open to changing it still.